go/cmd/dolt/commands/cherry-pick.go

```diff
@@ -0,0 +1,262 @@
+// Copyright 2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"context"
+
+	"github.com/dolthub/dolt/go/libraries/doltcore/merge"
+	"github.com/dolthub/dolt/go/store/hash"
+
+	"github.com/dolthub/dolt/go/cmd/dolt/cli"
+	"github.com/dolthub/dolt/go/cmd/dolt/errhand"
+	eventsapi "github.com/dolthub/dolt/go/gen/proto/dolt/services/eventsapi/v1alpha1"
+	"github.com/dolthub/dolt/go/libraries/doltcore/doltdb"
+	"github.com/dolthub/dolt/go/libraries/doltcore/env"
+	"github.com/dolthub/dolt/go/libraries/doltcore/table/editor"
+	"github.com/dolthub/dolt/go/libraries/utils/argparser"
+)
+
+var cherryPickDocs = cli.CommandDocumentationContent{
+	ShortDesc: `Apply the changes introduced by an existing commit.`,
+	LongDesc: `
+Updates tables in the clean working set with changes introduced in cherry-picked commit and creates a new commit with applied changes.
+
+Currently, schema changes introduced in cherry-pick commit are not supported. Row data changes are allowed with a table schema in working set 
```

Comment:
```suggestion
Cherry-picking merge commits or commits with schema changes is not currently supported. Row data changes are allowed as long as the two table schemas are exactly identical. 
```

(Could remove that next line, too)